app/services/service_agreement_creator.rb

```diff
@@ -1,15 +1,17 @@
 # frozen_string_literal: true
 
 class ServiceAgreementCreator
-  attr_reader :service_agreement
+  attr_reader :service_agreement, :civil_servant
 
-  def initialize(organization_admin)
+  def initialize(organization_admin, params)
```

Comment:
That's really weird that you expect params in initialize and in call, that are basically the same. 
Can you please take organization_admin and civil_servant as attributes in initialize?
So you would have a before_action :load_civil_servant in controller which initializes or finds a civil servant and pass it to the service